testsuite/classlibrary/TestUGen_RTAlloc.sc

```diff
@@ -1,326 +1,322 @@
 TestUGen_RTAlloc : UnitTest {
 
-	var server;
+	classvar server;
 
-	setUp {
-		server = Server(this.class.name);
-		server.options.memSize = 1024;
+	*initClass {
+		passVerbosity = UnitTest.brief;
+	}
+
+	*setUpClass {
+		server = Server(this.name);
+		server.options.sampleRate = 48000;
 		server.options.blockSize = 64;
-		this.bootServer(server);
+		server.options.memSize = 2 ** 13; // scsynth default
+		// - tests fail with memSize < 256 (GVerb fails)
+		// - testing with memSize >= 2 ** 20 would require multiple allocations
+		//   (allocation size itself can overflow size_t for some UGens)
+		server.bootSync;
 	}
 
-	tearDown {
+	*tearDownClass {
 		server.quit.remove;
 	}
 
-	test_allocFail_clearUnit {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(DC.ar(1), 100); }.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
-			cond.signalOne();
-		});
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 0.0, "a failed RTAlloc should clear unit outputs")
-	}
+	// Async helpers
 
-	test_allocFail_continueProcessing_sameNode {
-		var cond = CondVar(), output = -1;
-		{ DelayN.ar(Silent.ar, 100); DC.ar(1)}.loadToFloatArray(1 / server.sampleRate, server, {|val|
-			output = val[0];
+	awaitSynthOutput { |synthFn, runtimeSamples=(server.options.blockSize), timeout=1|
+		var cond = CondVar(), synthOutput = nil;
+		var runtimeSeconds = runtimeSamples / server.sampleRate;
+		// allow extra timeout for longer tests (e.g. testFFT with big memSize)
+		if (runtimeSeconds > timeout) { 
+			timeout = runtimeSeconds + timeout
+		};
+		synthFn.loadToFloatArray(runtimeSeconds, server) { |out|
+			synthOutput = out;
 			cond.signalOne();
-		});
-
-		cond.waitFor(1, { output != -1 });
-		this.assertFloatEquals(output, 1.0, "a failed RTAlloc should not block other units in the same node")
+		};
+		cond.waitFor(timeout) { synthOutput.notNil };
```

Comment:
Might be worth putting an assert here (cond.waitforn returns a bool) checking we didn't timeout as not all tests check the output is nil?

Reply:
I've added the missing timeout checks. I tried to integrate them in awaitSynthOutput, but in the end I thought it was more readable if done explicitly in every test. Otherwise tests would have to check `synthOutput.notNil` anyways, or they would run the other (non timed-out) assert.

Reply:
Thanks!

One interesting corner case could be if the timeout expires, but before the waiting thread is resumed, the thread that sets the output completes. Your approach would not count this as an error.... but this seems inconsequential because the data would be there.

Reply:
Yes, in that case the test fails with a "test should complete" message, signaling that anyways the test took too long to complete, and that's because something wrong happened. This happens for example if some test crashes the server, causing all subsequent tests to fail... maybe we should have this covered too? For example by waiting for server to boot if it's not booted already (mostly applies only to test_allUGens test)